src/Symfony/Component/Translation/DoctrineMessageCatalogue.php

```diff
@@ -0,0 +1,218 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation;
+
+use Doctrine\Common\Cache\Cache;
+use Doctrine\Common\Cache\MultiGetCache;
+
+/**
+ * @author Abdellatif Ait boudad <a.aitboudad@gmail.com>
+ */
+class DoctrineMessageCatalogue extends MessageCatalogue
+{
+    const PREFIX = 'sf2_translation_';
+    const CATALOGUE_DOMAINS = 'domains';
+    const CATALOGUE_DOMAIN_METADATA = 'domain_meta_';
+
+    /**
+     * @var Cache
+     */
+    private $cache;
+
+    /**
+     * @var string
+     */
+    private $prefix;
+
+    /**
+     * @var array
+     */
+    private $domains = array();
+
+    /**
+     * @param string $locale
+     * @param Cache  $cache
+     * @param string $prefix
+     */
+    public function __construct($locale, Cache $cache, $prefix = self::PREFIX)
+    {
+        parent::__construct($locale);
+        if (0 === strlen($prefix)) {
+            throw new \InvalidArgumentException('$prefix cannot be empty.');
+        }
+
+        $this->cache = $cache;
+        $this->prefix = $prefix.$locale;
```

Comment:
any place using the prefix is adding a ``_`` after it. I suggest adding it here already